pkg/controller/ingressconfig/ingressconfig_controller.go

```diff
@@ -46,8 +48,53 @@ func newIngressConfigReconciler(client crclient.Client, status *statusmanager.St
 	return &ReconcileIngressConfigs{client: client, status: status}
 }
 
+// isIngressCapabilityEnabled checks if the Ingress capability is enabled in the cluster
+func isIngressCapabilityEnabled(client crclient.Client) (bool, error) {
+	hcpCfg := hypershift.NewHyperShiftConfig()
+	return isIngressCapabilityEnabledWithConfig(client, hcpCfg)
+}
+
+// isIngressCapabilityEnabledWithConfig checks if the Ingress capability is enabled in the cluster
+func isIngressCapabilityEnabledWithConfig(client crclient.Client, hcpCfg *hypershift.HyperShiftConfig) (bool, error) {
+	// Handle nil client
+	if client == nil {
+		return false, fmt.Errorf("client cannot be nil")
+	}
+
+	// Check if this is a hypershift cluster
+	if !hcpCfg.Enabled {
+		// For non-hypershift clusters, assume ingress capability is always enabled
+		return true, nil
+	}
+
+	// For hypershift clusters, check if the IngressController CRD exists
+	gvk := operv1.SchemeGroupVersion.WithKind("IngressController")
+	mapping, err := client.RESTMapper().RESTMapping(gvk.GroupKind(), gvk.Version)
+	if err != nil {
```

Comment:
This should actually check if the error is what you expect for a missing rest mapping and if it is not the function should return an error.